src/main/java/net/fabricmc/api/FakeModProvider.java

```diff
@@ -0,0 +1,13 @@
+package net.fabricmc.api;
+
+import net.fabricmc.loader.api.ModContainer;
+
+import java.util.Collection;
+
+/**
+ * Hook for adding "fake" mod containers to Loader's list.
+ * Should run <i>after</i> {@link ModInitializer#onInitialize()}
+ */
+public interface FakeModProvider {
+	Collection<ModContainer> getFakeMods();
```

Comment:
Should this be like below instead?

```java
void registerFakeMods(Consumer<ProvidedModContainer> acceptor);
```

That would allow for a few things:
No need to worry about the whole collection being passed as null.
Anyone using the method would do the following for example:
```java
public void registerFakeMods(Consumer<ProvidedModContainer> acceptor) {
    ProvidedModContainer container1 = ...
    ProvidedModContainer container2 = ...
    acceptor.accept(container1);
    acceptor.accept(container2);
}
```

Reply:
that might be a good idea. I'll think about it.